### ISSUE TITLE
aws_route_table input validation and fallback values

### ISSUE THREAD

**Issue by AFriemann** [hashicorp/terraform-provider-aws#13348]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.12.24
+ provider.aws v2.62.0
```

### Affected Resource(s)

* aws_route_table

### Terraform Configuration Files

```hcl
variable "vpc_id" {
  type = string
}

variable "nat_gateway" {
  type = string
}

variable "additional_routes" {
  description = "Additional routes."
  type = map(object({
    egress_only_gateway_id    = string
    gateway_id                = string
    instance_id               = string
    ipv6_cidr_block           = string
    nat_gateway_id            = string
    network_interface_id      = string
    transit_gateway_id        = string
    vpc_peering_connection_id = string
  }))
  default = {}
}

resource "aws_route_table" "private" {
  vpc_id = var.vpc_id

  route {
    cidr_block                = "0.0.0.0/0"
    egress_only_gateway_id    = ""
    gateway_id                = ""
    instance_id               = ""
    nat_gateway_id            = var.nat_gateway
    network_interface_id      = ""
    transit_gateway_id        = ""
    vpc_peering_connection_id = ""
  }

  dynamic "route" {
    for_each = var.additional_routes

    content {
      cidr_block                = route.key
      egress_only_gateway_id    = route.value.egress_only_gateway_id
      gateway_id                = route.value.gateway_id
      instance_id               = route.value.instance_id
      ipv6_cidr_block           = route.value.ipv6_cidr_block
      nat_gateway_id            = route.value.nat_gateway_id
      network_interface_id      = route.value.network_interface_id
      transit_gateway_id        = route.value.transit_gateway_id
      vpc_peering_connection_id = route.value.vpc_peering_connection_id
    }
  }
}
```

### Debug Output

I'm omitting this since it contains a lot of potentially sensitive data. I think the issue is pretty clear in terms of code though.

### Expected Behavior

Let's say we run this with variables set like this

```hcl
vpc_id = "vpc-12345"
nat_gateway = "nat-12345"

additional_routes = {
  "10.10.10.0/24" = {
    egress_only_gateway_id    = null
    gateway_id                = null
    instance_id               = null
    ipv6_cidr_block           = null
    nat_gateway_id            = null
    network_interface_id      = null
    transit_gateway_id        = null
    vpc_peering_connection_id = "pcx-12345"
  },
  "127.0.0.0/16" = {
    egress_only_gateway_id    = null
    gateway_id                = null
-   instance_id               = "i-12345"
+   instance_id               = "i-12345678"
    ipv6_cidr_block           = null
    nat_gateway_id            = null
    network_interface_id      = null
    transit_gateway_id        = null
    vpc_peering_connection_id = null
  }
}
```

where one of those routes was changed.

This should only cause the affected route to change.

### Actual Behavior

Since unavailable routes are handled with empty strings internally, this will cause all routes to be recreated since `"" != null`.

Setting all unused values to `""` however will result in a CIDR validation failure on the ipv6_cidr_block input.

This effectively means that all routes are always re-created when one of the inputs changes.

### References

ipv_cidr_block input validation: https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_route_table.go#L61

**Comment by ewbankkit** [comment]
Related:
* https://github.com/terraform-providers/terraform-provider-aws/issues/13003

**Comment by ewbankkit** [comment]
The `null` vs. `"` diffs can be resolved once we address the breaking change in https://github.com/hashicorp/terraform-provider-aws/issues/14197.
The CIDR validation was addressed via https://github.com/hashicorp/terraform-provider-aws/pull/13778, released in **v2.68.0** of the AWS Provider.
